### PR TITLE
Move cards to the store

### PR DESCRIPTION
**Pull request by kvikshaug**
As part of the designs/jobs visualizations.

### CODE REVIEW

File: src/views/InteractiveMap/Card.vue [hunk 1]
```diff
@@ -181,27 +181,46 @@ export default Vue.extend({
       immediate: true,
       handler() {
         // Reset all modifications when the selected model changes.
-        this.card.reactionAdditions = [];
-        this.card.reactionKnockouts = [];
-        this.card.geneKnockouts = [];
-        this.card.editedBounds = [];
+        this.$store.commit("interactiveMap/updateCard", {
```
Comment (Midnighter) [inline src/views/InteractiveMap/Card.vue]: I would map this mutation into a local method as described in https://vuex.vuejs.org/guide/mutations.html#committing-mutations-in-components

File: src/store/modules/interactiveMap.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    addCard(state, card) {
+      state.cards.push(card);
+    },
+    removeCard(state, card) {
+      state.cards.splice(state.cards.indexOf(card), 1);
+    },
+    updateCard(state, { uuid, props }) {
+      const card = state.cards.find(c => c.uuid === uuid);
+      Object.keys(props).forEach(key => {
+        card[key] = props[key];
```
Comment (lukaw3d) [inline src/store/modules/interactiveMap.ts]: I don't think Vuex can detect these (and other) deep object changes (e.g. to re-run dependent `computed` properties on components)

Recommendation here: https://vuex.vuejs.org/guide/mutations.html#mutations-follow-vue-s-reactivity-rules

Comment (kvikshaug) [reply src/store/modules/interactiveMap.ts]: That's indeed the case when you add new properties to an object. Here we're just modifying existing properties though, and that should still be detected by deep watchers. Regardless, it's probably good practice to just use `Vue.set` in general, so I'll update this - thanks.

File: src/views/InteractiveMap/Card.vue (hunk 1 above)
Comment (kvikshaug) [reply src/views/InteractiveMap/Card.vue]: That's a nice improvement, thanks :+1:

**Pull request merged**